js/components/badges/modal.vue

```diff
@@ -80,8 +80,8 @@ export default {
         }
     },
     methods: {
-        confirm: function() {
-            let to_add = this.selected.filter((badge) => {
+        confirm() {
+            const to_add = this.selected.filter((badge) => {
```

Comment:
I wonder if `const to_add = !this.selected.filter(this.initial.indexOf)` would work :)